File: backend/internal/ast/ast.go
package ast

import "app/internal/statement"

type Program struct {
	statements []statement.Statement
}

func NewProgram() *Program {
	return &Program{
		statements: make([]statement.Statement, 0),
	}
}

func (program *Program) AppendStatement(stmt statement.Statement) {
	program.statements = append(program.statements, stmt)
}

func (program *Program) Get() *[]statement.Statement {
	return &program.statements
}

File: backend/internal/ast/ast.py
from backend.internal.statements import Statement


class Program:
    def __init__(self) -> None:
        self._stmts: list[Statement] = []

    def append(self, stmt: Statement) -> None:
        self._stmts.append(stmt)

    def get(self) -> list[Statement]:
        return self._stmts
